lib/improver/tests/wind_components/test_ResolveWindComponents.py

```diff
@@ -0,0 +1,203 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2018 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for the wind_components.ResolveWindComponents plugin."""
+
+import unittest
+import numpy as np
+from cf_units import Unit
+
+import iris
+from iris.tests import IrisTest
+from iris.coords import DimCoord
+from iris.coord_systems import OSGB
+
+from improver.wind_components import ResolveWindComponents
+
+
+def set_up_cube(data_2d, name, unit):
+    """Set up a 2D test cube of wind direction or speed"""
+    x_coord = DimCoord(np.arange(data_2d.shape[1]), "projection_x_coordinate",
+                       units="metres")
+    y_coord = DimCoord(np.arange(data_2d.shape[0]), "projection_y_coordinate",
+                       units="metres")
+    cube = iris.cube.Cube(data_2d, standard_name=name, units=unit,
+                          dim_coords_and_dims=[(y_coord, 0), (x_coord, 1)])
+    time_unit = Unit("hours since 1970-01-01 00:00:00", "gregorian")
+    t_coord = DimCoord(402292.5, "time", units=time_unit)
+    cube.add_aux_coord(t_coord)
+    return cube
+
+
+def expand_realizations(cube, nreal):
+    """Add realization dimension with nreal points by copying cube data"""
```

Comment:
Since the data is wind-speed + direction on height would that be a better additional coordinate?